src/tabs/serverAdmins/ManagePermissionsDialog.tsx

```diff
@@ -0,0 +1,104 @@
+
+  public static show(adminId: string, okCallback: (adminId: string, newPermissions: AdminPermissions) => void,
+                     adminPermissions: AdminPermissions) {
+    RX.Modal.show(
+      <ManagePermissionsDialog adminId={adminId} okCallback={okCallback} adminPermissions={adminPermissions} />, "managePermissionsDialog",
+    );
+  }
+
+  public render() {
+    return (
+      <RX.View style={Styles.whiteBox}>
+        <RX.Text>Admin Permission Management</RX.Text>
+        <CheckBox text={"Execute Console Cheat Commands"}
+                  onCheckedChange={(checked) => this.updatePermission(0, checked)}
+                  checkedByDefault={this.props.adminPermissions.permissions.consoleCheat}/>
+        <CheckBox text={"Execute Console User Management Commands"}
+                  onCheckedChange={(checked) => this.updatePermission(1, checked)}
+                  checkedByDefault={this.props.adminPermissions.permissions.consoleUserManagement}/>
+        <CheckBox text={"Execute Console Server Management Commands"}
+                  onCheckedChange={(checked) => this.updatePermission(2, checked)}
+                  checkedByDefault={this.props.adminPermissions.permissions.consoleServerManagement}/>
+        <CheckBox text={"Execute Console Debug Commands"}
+                  onCheckedChange={(checked) => this.updatePermission(3, checked)}
+                  checkedByDefault={this.props.adminPermissions.permissions.consoleDebug}/>
+        <CheckBox text={"Install Modules"}
+                  onCheckedChange={(checked) => this.updatePermission(4, checked)}
+                  checkedByDefault={this.props.adminPermissions.permissions.installModules}/>
+        <CheckBox text={"Create/Backup/Rename Games"}
+                  onCheckedChange={(checked) => this.updatePermission(5, checked)}
+                  checkedByDefault={this.props.adminPermissions.permissions.createBackupRenameGames}/>
+        <CheckBox text={"Stop Games"}
+                  onCheckedChange={(checked) => this.updatePermission(6, checked)}
+                  checkedByDefault={this.props.adminPermissions.permissions.startStopGames}/>
+        <CheckBox text={"Delete Games"}
+                  onCheckedChange={(checked) => this.updatePermission(7, checked)}
+                  checkedByDefault={this.props.adminPermissions.permissions.deleteGames}/>
+        <CheckBox text={"Change Settings"}
+                  onCheckedChange={(checked) => this.updatePermission(8, checked)}
+                  checkedByDefault={this.props.adminPermissions.permissions.changeSettings}/>
+        <CheckBox text={"Admin Management"}
+                  onCheckedChange={(checked) => this.updatePermission(9, checked)}
+                  checkedByDefault={this.props.adminPermissions.permissions.adminManagement}/>
+        <OkCancelButtonBar onOk={this.onOk} onCancel={this.onCancel} />
+      </RX.View>
+    );
+  }
+
+  private updatePermission(index: number, checked: boolean) {
```

Comment:
In my opinion this is a bit verbose and not very useful - having these numbers around can easily get confusing if you have to work with this code later, and could lead to weird bugs if you happen to mistype one of them when working with it.

I'd just move the various `this.props.adminPermissions.permissions.consoleCheat = checked` lines to the lambda functions which are bound to the `onCheckedChange` properties in the `render` method.
An alternative could also be replacing the numerical indices with an enum, or at least store them as constants (e.g. `const CHEAT_PERMISSION_ID = 1;`) - however this would just add unnecessary bloat IMHO.